packages/vaex-plotly/vaex/plotly/__init__.py

```diff
@@ -0,0 +1,359 @@
+        self.df = df
+
+    def scatter(self, x, y, xerr=None, yerr=None, selection=None,
+                size=None, color=None, symbol=None,
+                label=None, xlabel=None, ylabel=None, title=None,
+                colorbar=None, colorbar_label=None, colormap=None,
+                figure_height=None, figure_width=None,
+                tooltip_title=None, tooltip_data=None,
+                length_limit=50_000, length_check=True):
+        """Scatter plot using plotly.
+        Convenience wrapper around plotly.graph_objs.Scatter when for working with small DataFrames or selections.
+
+        :param x: Expression to plot on the x axis. Can be a list to plot multiple sets
+        :param y: Expression to plot on the y axis. Can be a list to plot multiple sets
+        :param xerr: Expression or a list of expressions for x-error bars
+        :param xerr: Expression or a list of expressions for y-error bars
+        :param selection: Selection, or None. Can be a list to plot multiple sets
+        :param size: The size of the markers. Can be an Expression or a list, if multiple sets are plotted
+        :param color: The color of the markers. Can be an Expression or a list if multiple sets are plotted
+        :param symbol: Plotly symbols for the markers. Can be a list if multiple sets are plotted
+        :param label: Label for the legend
+        :param xlabel: label for x axis, if None .label(x) is used
+        :param ylabel: label for y axis, if None .label(y) is used
+        :param title: the plot title
+        :param colorbar: if True, display a colorbar
+        :param colorbar_label: A label for the colorbar
+        :param colormap: A name of a colormap/colorscale supported by plotly
+        :param figure_height: The figure height in pix
+        :param figure_width: The figure width in pix
+        :param tooltip_title: Expression for the tooltip title
+        :param tooltip_data: A list of expressions for the extra tooltip data
+        :param length_limit: maximum number of rows it will plot
+        :param length_check: should we do the maximum row check or not?
+        :return plotly.graph_objs._figurewidget.FigureWidget fig: a plotly FigureWidget
+        """
+
+        import plotly.graph_objs as go
+
+        x = _ensure_list(x)
+        y = _ensure_list(y)
+
+        x = _ensure_strings_from_expressions(x)
+        y = _ensure_strings_from_expressions(y)
+        assert len(x) == len(y), 'x and y should have the same number of Expressions.'
+        num_traces = len(x)
+
+        args = self._arg_len_check(num_traces, xerr=xerr, yerr=yerr, size=size,
+                                   color=color, symbol=symbol, label=label,
+                                   selection=selection, tooltip_title=tooltip_title)
+        xerr, yerr, size, color, symbol, label, selection, tooltip_title = args
+
+        if length_check:
+            count = np.sum([self.df.count(selection=sel) for sel in selection])
```

Comment:
better pass selection as the list, since now it does multiple passes over the data

Reply:
you can argue that since it should be used with small data, it doesn't matter. However, this is at the check phase, so someone may want to plot a billion points, so at this point, it might be important.